node/node_test.go

```diff
@@ -760,6 +760,78 @@ func TestConfig_Preset(t *testing.T) {
 	})
 }
 
+func TestConfig_CustomTypes(t *testing.T) {
+	const name = "testnet"
+
+	tt := []struct {
+		name         string
+		cli          string
+		config       string
+		updatePreset func(*testing.T, *config.Config)
+	}{
+		{
+			name:   "smeshing-opts-provider",
+			cli:    "--smeshing-opts-provider=1337",
+			config: `{"smeshing": {"smeshing-opts": {"smeshing-opts-provider": "1337"}}}`,
```

Comment:
... then this will work:
```suggestion
			config: `{"smeshing": {"smeshing-opts": {"smeshing-opts-provider": 1337}}}`,
```